core/domain/collection_domain.py

```diff
@@ -1303,3 +1303,20 @@ def does_user_have_any_role(self, user_id):
             user_id in self.editor_ids or
             user_id in self.viewer_ids
         )
+
+    def add_new_contributor(self, contributor_id):
+        """Add a new contributor to the contributors summary.
+
+        Args:
+            contributor_id: str. ID of the contributor to be added.
+        """
+        contributors_summary = (
+            self.contributors_summary if self.contributors_summary else {})
+
+        if contributor_id not in constants.SYSTEM_USER_IDS:
+            self.contributors_summary[contributor_id] = (
+                self.contributors_summary.get(contributor_id, 0) + 1)
```

Comment:
What's the value represent in the dict contributors_summary?

Reply:
Not sure what you mean? The number for each user represents how many contributions the user made in the collection or exploration.

Reply:
As per offline discussion, we found that this method isn't adding a new contributor instead it's registering a contribution made by a user (irrespective of whether it's a new contributor or existing contributor) in the collection so we are going to rename the function to `add_contribution_by_user`.

Reply:
Done.